### PR TITLE
obtuse and confusing error with local feature under WSL #834

### PR DESCRIPTION
**Pull request by prathameshzarkar9**
[Issue 834](https://github.com/devcontainers/cli/pull/838)
Confusing error message modified to more understandable message.

### CODE REVIEW

File: src/spec-configuration/containerFeaturesConfiguration.ts
Comment (samruddhikhandale) [inline src/spec-configuration/containerFeaturesConfiguration.ts:701]: Can you follow the messaging in https://github.com/devcontainers/cli/issues/834#issuecomment-2148074401? Thanks! 

We need to append the new message and not replace it. The issue comment has the needed message string.

**Review by samruddhikhandale — CHANGES_REQUESTED** [review]

**Review by dotysan — APPROVED** [review]

File: src/spec-configuration/containerFeaturesConfiguration.ts
Comment (chrmarti) [inline src/spec-configuration/containerFeaturesConfiguration.ts:701]: - Since it is twice the same message, could you extract it to a local variable?
- There is indentation on the second line of the message that needs to be removed.
- You could use an actual newline in stead of `\n`.

**Review by chrmarti — CHANGES_REQUESTED** [review]
Thanks for the PR, left a comment.

File: src/spec-configuration/containerFeaturesConfiguration.ts
Comment (prathameshzarkar9) [reply src/spec-configuration/containerFeaturesConfiguration.ts:701]: @samruddhikhandale @chrmarti,
kindly check the modifications done to address this.

**Comment by prathameshzarkar9** [comment]
Hi @samruddhikhandale ,
I have addressed review comments and also modified the error message as suggested by @chrmarti .
Kindly review and approve the same.

File: src/spec-configuration/containerFeaturesConfiguration.ts
Comment (samruddhikhandale) [inline src/spec-configuration/containerFeaturesConfiguration.ts:689]:
```suggestion
	const errorMessage = `Legacy feature '${userFeatureId}' not supported. Please check https://containers.dev/features for replacements.
```

Comment (samruddhikhandale) [inline src/spec-configuration/containerFeaturesConfiguration.ts:689]: Can we move this variable to L703 ie inside [this](https://github.com/devcontainers/cli/pull/838/files#diff-93622e1db0f140d14a85a65ca0a4522090b385c3964de02011229c0523095470R702) condition?

File: src/spec-configuration/containerFeaturesConfiguration.ts
Comment (samruddhikhandale) [inline src/spec-configuration/containerFeaturesConfiguration.ts:690]:
```suggestion
If you were hoping to use local Features, remember to prepend your Feature name with "./". Please check https://containers.dev/implementors/features-distribution/#addendum-locally-referenced for more information.`;
```

Should we remove the tab?

File: src/test/cli.build.test.ts
Comment (samruddhikhandale) [inline src/test/cli.build.test.ts:37]: Thanks for adding the test!

Comment (samruddhikhandale) [inline src/test/cli.build.test.ts:37]: Nit: We can also assert for the entire error message.

**Review by samruddhikhandale — CHANGES_REQUESTED** [review]

File: src/spec-configuration/containerFeaturesConfiguration.ts
Comment (prathameshzarkar9) [reply src/spec-configuration/containerFeaturesConfiguration.ts:689]: done

Comment (prathameshzarkar9) [reply src/spec-configuration/containerFeaturesConfiguration.ts:689]: Done

File: src/spec-configuration/containerFeaturesConfiguration.ts
Comment (prathameshzarkar9) [reply src/spec-configuration/containerFeaturesConfiguration.ts:690]: tab removed

File: src/test/cli.build.test.ts
Comment (prathameshzarkar9) [reply src/test/cli.build.test.ts:37]: error message contains the dynamic feature name which is defined in the `devcontainer.json` file. Referred as `userFeatureId` where the error message is modified.
So used the match.

**Review by samruddhikhandale — APPROVED** [review]
Looks good to me, let's wait for another set of review from @chrmarti 

thanks!

**Review by chrmarti — APPROVED** [review]
LGTM, thanks!

**Pull request merged**
